packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/clientmodel/ClientMethod.java

```diff
@@ -836,6 +851,11 @@ public Builder hasWithContextOverload(boolean hasWithContextOverload) {
             return this;
         }
 
+        public Builder hidePageableParams(boolean hidePageableParams) {
```

Comment:
Unsure why we need a flag in ClientMethod for this behavior.

The `maxpagesize` is already under flag, and I assume the `token` related parameter should always be hidden?

Reply:
This is needed because we need to make the `token` param visible in "SinglePage" methods but not in methods that return "PagedIterable". So, always hiding would not work. We have to hide them only in aggregate paging methods and only in specific flavors.

Reply:
got it.

nit on the naming, `pageableParamsHidden` and `isPageableParamsHidden()`, as `is##` pattern seems more consistent in model classes.